src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -249,13 +261,26 @@ export class SavedObjectsClient {
       overwrite: options.overwrite,
     };
 
+    const currentWorkspaceId = this.currentWorkspaceId;
+    let finalWorkspaces;
+    if (options.hasOwnProperty('workspaces')) {
+      finalWorkspaces = options.workspaces;
+    } else if (typeof currentWorkspaceId === 'string') {
+      finalWorkspaces = [currentWorkspaceId];
```

Comment:
what's the difference of the currentworkspace id vs the one from the payload/options?

Reply:
The workspaces fields inside payload/options has a higher priority than currentWorkspaceId.

Reply:
Can't we just set workspace id into options when using the saved object client?

Reply:
There are 2 sources of workspaces:
- workspaces inside options.
- currentWorkspaceId in the instance.

the `workspaces inside options` has higher priority than the current workspaceId, if somewhere calls a method by specifying `workspaces inside options`, we will honor that and use exactly what they pass to the request payload.

We add `currentWorkspaceId in the instance` because actually there is few case that plugins want to overwrite the `workspaces inside options`, by using the `currentWorkspaceId in the instance`, dashboards page / visualizations page / data source management page will support workspace automatically without a single line of code changes because currentWorkspaceId is already set by workspace plugin when they call any CRUD operations through saved objects client.